src/client/interpreter/helpers.ts

```
@@ -79,23 +80,22 @@ export class InterpreterHelper implements IInterpreterHelper {
             }
         }
     }
+
     public async getInterpreterInformation(pythonPath: string): Promise<undefined | Partial<PythonEnvironment>> {
         const found = await this.pyenvs.getInterpreterInformation(pythonPath);
         if (found !== undefined) {
             return found;
         }
 
-        const fileHash = await this.hashProviderFactory
-            .create({ pythonPath })
-            .then((provider) => provider.getInterpreterHash(pythonPath))
-            .catch((ex) => {
-                traceError(`Failed to create File hash for interpreter ${pythonPath}`, ex);
-                return '';
-            });
+        const fileHash = await this.hashProvider.getInterpreterHash(pythonPath).catch((ex) => {
+            traceError(`Failed to create File hash for interpreter ${pythonPath}`, ex);
+            return '';
```

If getting hash fails, we should not cache the interpreter instead of using empty string as key for storing hash.